prospero-cli/src/test/java/org/wildfly/prospero/cli/CliMainTest.java

```diff
@@ -2,55 +2,41 @@
 
 import org.junit.Test;
 import org.junit.runner.RunWith;
-import org.mockito.Mock;
 import org.mockito.junit.MockitoJUnitRunner;
 
-import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
 
-@RunWith(MockitoJUnitRunner.class)
-public class CliMainTest {
 
-    @Mock
-    private CliMain.ActionFactory actionFactory;
+@RunWith(MockitoJUnitRunner.class)
+public class CliMainTest extends AbstractConsoleTest {
 
     @Test
-    public void errorIfArgNameDoesntStartWithDoubleHyphens() throws Exception {
-        try {
-            new CliMain(actionFactory).handleArgs(new String[]{"install", "dir=test"});
-            fail("Should have failed");
-        } catch (ArgumentParsingException e) {
-            assertEquals("Argument [dir=test] not recognized", e.getMessage());
-        }
+    public void errorIfArgNameDoesntStartWithDoubleHyphens() {
+        int exitCode = commandLine.execute("install", "dir=test");
+        assertEquals(ReturnCodes.INVALID_ARGUMENTS, exitCode);
+        assertTrue(getErrorOutput().contains("Missing required option: '--dir=<directory>'"));
     }
 
     @Test
-    public void errorIfArgumentHasNoValue() throws Exception {
-        try {
-            new CliMain(actionFactory).handleArgs(new String[]{"install", "--dir"});
-            fail("Should have failed");
-        } catch (ArgumentParsingException e) {
-            assertEquals("Argument value cannot be empty", e.getMessage());
-        }
+    public void errorIfArgumentHasNoValue() {
+        int exitCode = commandLine.execute("install", "--dir");
```

Comment:
Can you make the argument names use constants? Just to make it easier if/when we rename them.

Reply:
Yes, it was supposed to be in the `OptionNames` class, but then I deleted it to take a proper look at it later. I will add that back.